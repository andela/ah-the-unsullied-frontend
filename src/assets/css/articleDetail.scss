.description{
    color: gray;
}

.follow{
    border-color: #007CBE !important;
    color: #0971AB !important;
    background-color: aliceblue !important;
}

.tags-btn{
    margin-left: 10px;
    border-color: #007CBE !important;
    color: #0971AB !important;
    background-color: aliceblue !important;
    border-radius: 5% !important;
}

.prof-img{
    width: 50px;
    margin-left: 10px;
}

.bd-image img {
    width: 80% !important;
    height: 20% !important;
}
.danger {
    color: white; 
    background-color: Red !important;
    height: 30px;

}