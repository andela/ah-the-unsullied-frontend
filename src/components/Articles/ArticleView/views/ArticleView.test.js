import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';
import React from 'react';

import  articleDetail  from './ArticleView';

Enzyme.configure({ adapter: new Adapter() });

describe('articleDetail', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      auth: {
        isAuthenticated: true,
        user: {
          username: 'kwanj',
          email: 'kwanj@gmail.com'
        }
      },
      getArticles: jest.fn(() => {
        Promise.resolve();
      }),
      history: {
        push: jest.fn(() => {
          Promise.resolve();
        })
      },
      article: {
        article: {  
          article:{
          id: 6,
          author: {
            username: 'kwanj',
            image:
              'https://res.cloudinary.com/dadrqjrpw/image/upload/v1550732215/tzpffmzpmsy3lmijpauf.jpg',
            bio: null,
            following: null
          },
          body: '<p>sbsdbb</p>',
          title: 'how-to-become-a-god',
          description: 'First article from the frontens #theunsullied',
          rating: 0,
          tag_list: ['kwanj'],
          read_time: '1 min read',
          likes: 0,
          dislikes: 0,
          liked: false,
          disliked: false,
          slug: 'how-to-become-a-god-xd0ctx6n',
          created_at: '2019-02-28T10:04:42.687397Z',
          updated_at: '2019-02-28T10:15:29.020191Z',
          is_reported: false
        }
      }
    }
    };
    wrapper = shallow(<articleDetail {...props} />);
  });
  it('should render correctly', () =>{
    expect(wrapper).toMatchSnapshot()
  })

});
