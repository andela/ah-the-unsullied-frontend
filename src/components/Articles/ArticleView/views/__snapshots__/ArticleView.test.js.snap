// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articleDetail should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <articleDetail
    article={
      Object {
        "article": Object {
          "article": Object {
            "author": Object {
              "bio": null,
              "following": null,
              "image": "https://res.cloudinary.com/dadrqjrpw/image/upload/v1550732215/tzpffmzpmsy3lmijpauf.jpg",
              "username": "kwanj",
            },
            "body": "<p>sbsdbb</p>",
            "created_at": "2019-02-28T10:04:42.687397Z",
            "description": "First article from the frontens #theunsullied",
            "disliked": false,
            "dislikes": 0,
            "id": 6,
            "is_reported": false,
            "liked": false,
            "likes": 0,
            "rating": 0,
            "read_time": "1 min read",
            "slug": "how-to-become-a-god-xd0ctx6n",
            "tag_list": Array [
              "kwanj",
            ],
            "title": "how-to-become-a-god",
            "updated_at": "2019-02-28T10:15:29.020191Z",
          },
        },
      }
    }
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "email": "kwanj@gmail.com",
          "username": "kwanj",
        },
      }
    }
    getArticles={[MockFunction]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "article": Object {
        "article": Object {
          "article": Object {
            "author": Object {
              "bio": null,
              "following": null,
              "image": "https://res.cloudinary.com/dadrqjrpw/image/upload/v1550732215/tzpffmzpmsy3lmijpauf.jpg",
              "username": "kwanj",
            },
            "body": "<p>sbsdbb</p>",
            "created_at": "2019-02-28T10:04:42.687397Z",
            "description": "First article from the frontens #theunsullied",
            "disliked": false,
            "dislikes": 0,
            "id": 6,
            "is_reported": false,
            "liked": false,
            "likes": 0,
            "rating": 0,
            "read_time": "1 min read",
            "slug": "how-to-become-a-god-xd0ctx6n",
            "tag_list": Array [
              "kwanj",
            ],
            "title": "how-to-become-a-god",
            "updated_at": "2019-02-28T10:15:29.020191Z",
          },
        },
      },
      "auth": Object {
        "isAuthenticated": true,
        "user": Object {
          "email": "kwanj@gmail.com",
          "username": "kwanj",
        },
      },
      "getArticles": [MockFunction],
      "history": Object {
        "push": [MockFunction],
      },
    },
    "ref": null,
    "rendered": null,
    "type": "articleDetail",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "article": Object {
          "article": Object {
            "article": Object {
              "author": Object {
                "bio": null,
                "following": null,
                "image": "https://res.cloudinary.com/dadrqjrpw/image/upload/v1550732215/tzpffmzpmsy3lmijpauf.jpg",
                "username": "kwanj",
              },
              "body": "<p>sbsdbb</p>",
              "created_at": "2019-02-28T10:04:42.687397Z",
              "description": "First article from the frontens #theunsullied",
              "disliked": false,
              "dislikes": 0,
              "id": 6,
              "is_reported": false,
              "liked": false,
              "likes": 0,
              "rating": 0,
              "read_time": "1 min read",
              "slug": "how-to-become-a-god-xd0ctx6n",
              "tag_list": Array [
                "kwanj",
              ],
              "title": "how-to-become-a-god",
              "updated_at": "2019-02-28T10:15:29.020191Z",
            },
          },
        },
        "auth": Object {
          "isAuthenticated": true,
          "user": Object {
            "email": "kwanj@gmail.com",
            "username": "kwanj",
          },
        },
        "getArticles": [MockFunction],
        "history": Object {
          "push": [MockFunction],
        },
      },
      "ref": null,
      "rendered": null,
      "type": "articleDetail",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
